### DISCUSSION TITLE
General Xinput or DirectInput controller support

### DISCUSSION THREAD

**Discussion by qlum** [cathery/sys-con#29] [Controller requests]
So far I tested some pc controller (Logitech f510).
It seems it's not recognized, neither in Dinput mode nor Xinput mode. 

Xinput generally means emulating a xbox360 controller, so I thought that this may have worked but sadly not.

Would this be something worth looking into?

**Comment by cathery** [comment]
This is something I want to look into. If no setbacks show up I'll try testing it tomorrow

**Comment by cathery** [comment]
Can you launch this applet and send me a screenshot of your scan?
https://files.catbox.moe/re60qo.rar

**Comment by qlum** [comment]
Here you go:
Xinput:
![X-input](https://user-images.githubusercontent.com/3518815/68436007-13bfb900-01bd-11ea-964e-aced558db17c.jpg)

Direct Input:
![D-input](https://user-images.githubusercontent.com/3518815/68436006-13bfb900-01bd-11ea-9818-f69e99cd0eb0.jpg)

**Comment by cathery** [comment]
Thank you. I'll see what I can do

**Comment by cathery** [comment]
Can you see if XInput works on this version? https://files.catbox.moe/5zbqou.rar

**Comment by developersu** [comment]
Mmm.. created PR for this. X-input works awesome already %)

**Comment by qlum** [comment]
This seems to work specifically in xinput mode, via usb-otg in handheld mode. This is also how I posted the screenshot
Here is a screenshot of the same device in docked mode, where it does not work.

![screensot](https://user-images.githubusercontent.com/3518815/68506170-46bf8680-0269-11ea-89f7-b09469419fc1.jpg)


I also tried using my keyboard, which also doubles as an xinput controller. Sadly that also doesnt't work. Here is a screenshot of that too for good measure (please note not all devices are shown as only so much fit on screen and I saw no way to show the rest):

![2019110820472900-DB1426D1DFD034027CECDE9C2DD914B8](https://user-images.githubusercontent.com/3518815/68506299-a158e280-0269-11ea-95cf-a6c58d5109af.jpg)

**Comment by qlum** [comment]
I want to add that version 0.4.0 seems to work for xinput on multiple controllers. (tested on the logitech f510 and the 8bitdo sn30 pro (Which is compatible with the switch anyway. So no real reason to use xinput other than to test it.

My keyboard still does not work, It is listed as a xbox360 wireless receiver for Windows. As well as 2 keyboards. Maybe that creates a conflict or there is some other reason why it does not work. 

Direct input mode on my controllers also does not work yet but that is to be expected as direct input is not supported.

**Comment by ybgmoli** [comment]
Hi. I have a GameSir T1s controller, when I put it in Ximput mode, the console recognizes it perfectly using syscon.

**Comment by DrSkunk** [comment]
Currently testing a generic Xinput controller that when watching the controller list briefly pops up for a fraction of second and then dissapears. This controller shows up as an Xbox 360 controller on Windows, so I assume it uses Xinput. But the packaging shows it has compatibility with PS3, so I think the driver switches the mode depending on the connected system.
![2020022918580100-DB1426D1DFD034027CECDE9C2DD914B8](https://user-images.githubusercontent.com/1058174/75612750-c6c67080-5b26-11ea-9a3b-321404127b86.jpg)
